cli/azd/pkg/infra/provisioning/terraform/terraform_provider.go

```diff
@@ -452,6 +490,75 @@ func (t *TerraformProvider) createDeployment(ctx context.Context, modulePath str
 	return &template, nil
 }
 
+// collectAzureResources collects the set of resources from the root module of a terraform state file, including
+// resources from all child modules. Only resources managed by azure providers are considered (today, that's
+// just resources from the `registry.terraform.io/hashicorp/azurerm` provider). Only "managed" resources are
+// considered.
+func (t *TerraformProvider) collectAzureResources(rootModule terraformRootModule) []Resource {
+	// the set of resources we've seen (keyed by their id)
+	azureResources := make(map[string]struct{})
+
+	// Walk over all the modules (starting at the root) and mark each resource we see from the azure
+	// provider.
+	visitResource := func(r terraformResource) {
+		if r.Mode == terraformModeManaged && r.ProviderName == "registry.terraform.io/hashicorp/azurerm" {
+			if id, err := t.getIdForManagedResource(r); err != nil {
+				log.Printf("error determining id for resource: %v, ignoring...", err)
+			} else {
+				azureResources[id] = struct{}{}
+			}
+		}
+	}
+
+	var visitChildModule func(c terraformChildModule)
+	visitChildModule = func(c terraformChildModule) {
+		for _, r := range c.Resources {
+			visitResource(r)
+		}
+
+		for _, c := range c.ChildModules {
+			visitChildModule(c)
+		}
+	}
+
+	for _, r := range rootModule.Resources {
+		visitResource(r)
+	}
+
+	for _, c := range rootModule.ChildModules {
+		visitChildModule(c)
+	}
+
+	// At this point, allResources contains the ids of all the resources we discovered.
+	resources := make([]Resource, 0, len(azureResources))
+	for id := range azureResources {
+		resources = append(resources, Resource{
+			Id: id,
+		})
+	}
+
+	return resources
+}
+
+func (t *TerraformProvider) getIdForManagedResource(r terraformResource) (string, error) {
+	// Most azure resources use "id" as the key in the values bag that holds the resource id. However, some
+	// resources are special and use a different key for the Azure Resource Id.
```

Comment:
Slightly unfortunate that we'll have to keep this translation up to date with `tf`. I'm guessing there's no master list for this?

Reply:
Agree - I opened #868 to see if there is something automatic we can do here.